### ISSUE TITLE
Add support for Docker runtime args

### ISSUE THREAD

**Issue by Naninani** [MicrosoftDocs/vscodespaces#732]
Passing runtime arguments to the docker container when created is a must needed feature. We want to be able to set specific parameter is the config file such as:
    "--cpus=",
    "--memory=",
    "--memory-swap=",
    "--shm-size=",
    "--add-host=<some host>:127.0.0.1",
    "-v /usr/local/share/.cache:/usr/local/share/.cache"

Currently that is not supported.

We need to be able to pass those params so that we can streamline the local dev and lab envs.
Adding volumes mounts between the host and the container is probably one of the most critical for us, as we want to share the yarn cache and have it persistent on the host so that the user dont have to fetch all the deps again.

**Comment by lostintangent** [comment]
@jkeech @chuxel Thoughts? We support `runArgs` today, but are we tracking any improvements that would help here?

**Comment by jkeech** [comment]
We have an allow list of specific `runArgs` that we allow today, but those could be expanded as needed. For resource limits, I don't think we are going to fully open that up as we plan to add resource limits to the main containers by default to prevent issues with resource starvation causing connection drops.

**Comment by Chuxel** [comment]
Yeah we may want to switch to a "disallow list" instead.  There's actually a problem with running Chrome without setting `--ipc=host` unless you add additional arguments to the chrome startup.  I work around that in the desktop script by using an alias.

The "mounts" property allows for mounting volumes, but bind mounting is ignored currently.
